Inject null checks after base call in constructors

shana opened the pull request:
Fixes #51 

I've updated the tests as best I could, I'm running VS 2015 and the compiler is outputting different IL for certain things so I can't be certain that I've caught all the instances of classes with constructors calling base (or having base calls injected by the compiler).

ulrichb commented:
:warning: Please see https://github.com/Fody/NullGuard/issues/51#issuecomment-106851250 before merging this.

shana commented:
I changed the behaviour so it injects null checks before the base call only if the argument in question is used or accessed in some way in the base/this call itself. Forwarded arguments and arguments that aren't used in base/this calls get their null checks after the base/this call.

I also added some constructors to the SimpleClass and updated the relevant approved.txt files with the added IL (and fixed line numbers)

distantcam commented:
Closed per #51
